test/CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+project(Template_test)
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS ON)
+
+set(external_libs)
+if (NOT Template_IS_TOP_LEVEL)
+	FetchContent_Declare(Template
+			GIT_REPOSITORY https://github.com/LecrisUT/CMake-Template
+			GIT_TAG main
+			FIND_PACKAGE_ARGS CONFIG
+	)
+	list(APPEND external_libs Template)
+endif ()
+FetchContent_Declare(Catch2
+		GIT_REPOSITORY https://github.com/catchorg/Catch2
+		GIT_TAG v3.4.0
+		FIND_PACKAGE_ARGS CONFIG
+)
+list(APPEND external_libs Catch2)
+
+FetchContent_MakeAvailable(${external_libs})
+
+enable_testing()
+include(Catch)
+
+add_executable(test-suite)
+target_link_libraries(test-suite PRIVATE Catch2::Catch2WithMain Template::Template)
+
+function(Template_add_test test)
+	#[===[.md
+	# Template_add_test
+
+	Internal helper for adding functional tests specific for the current template project
+
+	## Synopsis
+	```cmake
+	Template_add_test(<name>
+			[TEST_NAME <test_name>]
+			[TARGET <target>]
+			[LABELS <label1> <label2>])
+	```
```

Comment:
I find myself often having to define such a wrapper for running basically functional tests, either to test packaging, or to test CMake helerp functions